test/mciproject.yml

```diff
@@ -99,6 +106,38 @@ tasks:
           display_name: Binaries
           remote_file: wiredtiger/${build_variant}/${revision}/binaries/${build_id}.tgz
 
+  - name: csuite-test
+    depends_on: []
+    commands:
+      - func: "fetch source"
+      - command: git.apply_patch
+        params:
+          directory: wiredtiger
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger"
+          script: |
+            set -o errexit
+            set -o verbose
+            if [ "Windows_NT" = "$OS" ]; then
+              scons.bat --enable-python=c:\\swigwin-3.0.2\\swig.exe --enable-diagnostic ${smp_command|}
+            else
+              ./build_posix/reconf
+              ${configure_env_vars|} ./configure --enable-diagnostic --enable-python --enable-zlib --enable-strict
+              ${make_command|make} ${smp_command|} 2>&1
```

Comment:
I was expecting to have a compile job, that just compiles, then other jobs that copies the artifacts to run tests. I understand that compile is pretty fast for standalone WiredTiger, but it's more idiomatic method for Evergreen.

Reply:
That's actually the next thing I planned to do :) 
The only concern I have is the overhead of s3 put (for the compile task) and s3 get (for all other dependent tasks). But will test and see.